examples/compose/vtcompose.go

```diff
@@ -0,0 +1,566 @@
+
+	topologyFlags = flag.String("topologyFlags",
+		"-topo_implementation consul -topo_global_server_address consul1:8500 -topo_global_root vitess/global",
+		"Vitess Topology Flags config")
+	webPort   = flag.String("webPort", "8080", "Web port to be used")
+	gRpcPort  = flag.String("gRpcPort", "15999", "gRPC port to be used")
+	mySqlPort = flag.String("mySqlPort", "15306", "mySql port to be used")
+	cell      = flag.String("cell", "test", "Vitess Cell name")
+	keyspaceData = flag.String("keyspaces", "test_keyspace:2:1:create_messages.sql,create_tokens.sql:lookup_keyspace unsharded_keyspace:0:0:create_dinosaurs.sql,create_eggs.sql", "List of keyspace_name:num_of_shards:num_of_replica_tablets:schema_files:<optional>lookup_keyspace_name")
+)
+
+type keyspaceInfo struct {
+	keyspace string
+	shards  int
+	replicaTablets int
+	lookupKeyspace string
+	useLookups bool
+	schemaFile *os.File
+	schemaFileNames []string
+}
+
+func newKeyspaceInfo(keyspace string, shards int, replicaTablets int, schemaFiles []string, lookupKeyspace string) keyspaceInfo {
+	k := keyspaceInfo{keyspace: keyspace, shards: shards, replicaTablets: replicaTablets, schemaFileNames: schemaFiles, lookupKeyspace: lookupKeyspace}
+	if len(strings.TrimSpace(lookupKeyspace)) == 0 {
+		k.useLookups = false
+	} else {
+		k.useLookups = true
+	}
+
+	k.schemaFile = nil
+	return k
+}
+
+func main() {
+	flag.Parse()
+	keyspaceInfoMap := make(map[string]keyspaceInfo)
+
+	for _,v:= range strings.Split(*keyspaceData, " ") {
+		tokens := strings.Split(v, ":")
+		shards, _ := strconv.Atoi(tokens[1])
+		replicaTablets, _ := strconv.Atoi(tokens[2])
+		schemaFileNames := strings.Split(tokens[3], ",")
+		print(shards)
+
+		if len(tokens) > 4 {
+			keyspaceInfoMap[tokens[0]] = newKeyspaceInfo(tokens[0], shards, replicaTablets, schemaFileNames, tokens[4])
+		} else {
+			keyspaceInfoMap[tokens[0]] = newKeyspaceInfo(tokens[0], shards, replicaTablets, schemaFileNames, "")
+		}
+	}
+
+	for _, v := range keyspaceInfoMap {
+		v.schemaFile = createFile(fmt.Sprintf("tables/%s_schema_file.sql", v.keyspace))
+		appendtoSqlFile(v.schemaFileNames, v.schemaFile)
+		closeFile(v.schemaFile)
+	}
+
+	// Vschema Patching
+	for _, keyspaceData := range keyspaceInfoMap {
+		vSchemaFile := readFile(*baseVschemaFile)
+		if keyspaceData.shards == 0 {
+			vSchemaFile = applyJsonInMemoryPatch(vSchemaFile,`[{"op": "replace","path": "/sharded", "value": false}]`)
+		}
+
+		vSchemaFile, primaryTableColumns :=  addTablesVschemaPatch(vSchemaFile, keyspaceData.schemaFileNames)
+
+		if keyspaceData.useLookups {
+			lookupKeyspace := keyspaceInfoMap[keyspaceData.lookupKeyspace]
+			vSchemaFile = addLookupDataToVschema(vSchemaFile, lookupKeyspace.schemaFileNames, primaryTableColumns, lookupKeyspace.keyspace)
+		}
+
+		writeVschemaFile(vSchemaFile, fmt.Sprintf("%s_vschema.json", keyspaceData.keyspace))
+	}
+
+	// Docker Compose File Patches
+	dockerFile := readFile(*baseYamlFile)
```

Comment:
This variable name is a little misleading, as `Dockerfile` has a specific, different meaning

Reply:
Changed 👍